### DISCUSSION TITLE
Calibration graph "Intercept" and "Raw value" axis values use mg/dl values and ignore mmol/l settings.

### DISCUSSION THREAD

**Discussion by sashasochka** [NightscoutFoundation/xDrip#2983] [Ideas]
Hello, it would be more convenient to understand calibration values if they respected mmol/l settings on the Calibration Graph screen. Should also be very easy to implement. Would really appreciate.

**Comment by Navid200** [comment]
I am surprised.
I will be testing this when I return home.
If I don't get back to you in this thread in two weeks, please remind me.

**Comment by Navid200** [comment]
OK

This is what I see and understand.
The calibration graph has an x axis that corresponds to the raw values and a y axis that corresponds to the calibrated glucose readings.

If you change your units, the vertical (y) axis unit changes to correspond to the unit you have chosen.
The horizontal (x) axis units are independent of the chosen units because the raw values are always in mg/dL.

Please tell me if you have any questions.

**Reply by Navid200** [reply]
I may be understanding what you are suggesting.
I see that the intercept point and raw axis are independent of the selected units.
Regardless of what unit you choose, the calibration intercept point and the x axis corresponds to mg/dL.

I don't see a problem with doing what you have suggested.  But, I need to know why it is done this way.

@tzachi-dar 
Can I ask you why we have left the intercept point and raw values in the same mg/dL unit regardless of what unit the user chooses?

**Reply by tzachi-dar** [reply]
I don't think it has a good reason. Probably the one programing this did not think about it.

**Reply by sashasochka** [reply]
Just confirming that you are understanding correctly what I was trying to describe.

**Reply by Navid200** [reply]
We have some mechanisms in place that verify that the intercept does not violate a safe limit.  And this is in mg/dL.
So, we didn't want someone to be able to calibrate such that a very low reading would show as a high reading due to a calibration error resulting in them deciding to take insulin when they shouldn't.
The intercept is never allowed to go over a limit.  If it does, the calibration is rejected.

If we make this change, it could possibly cause some complications.  If the user changes their unit to mmol/L and erroneously think that the intercept limit is the same figure, it would be bad if they force the intercept.
The only way we can change this is if we hear from everyone who decided to leave this as is.  I wonder if it was left like this for exactly this reason.

**Reply by sashasochka** [reply]
Wouldn't it be useful to at least add the units to all labels to prevent any erroneous thinking, including the potential situation described?

**Reply by Navid200** [reply]
@sashasochka 
What exactly do you mean by "all labels"?

There is information I can add to the guide.  Not everything can be explained in detail in the app itself.  That's the case with items other than calibration also.

Please explain the scenario you are concerned about where the current calibration graph as is could cause a confusion for someone who uses xDrip as their CGM needing to calibrate on a regular basis.

From my perspective, the user enters the calibration in their chosen unit.
The unit is shown on the vertical axis on the calibration graph.
The horizontal axis represents the raw values that xDrip deals with, not the user.

**Reply by sashasochka** [reply]
I meant adding "mmol/l" or "mg/dl" to any number or respective label on the chart to avoid any source of potential confusion. E.g. to the "Raw value" label at the bottom of the calibration chart. Well, I had some quirky sensors lately and had a tough time calibrating them so I was looking at the calibration graph pretty often trying to understand the curve fit and it was a bit inconvenient having to do the conversions in my mind. Is there any other use case why anyone would look at the calibration chart? (And not prefer the values to be in mmol/l if that's their unit of choice)

**Reply by Navid200** [reply]
> Is there any other use case why anyone would look at the calibration chart? (And not prefer the values to be in mmol/l if that's their unit of choice)

The unit of the vertical axis is mmol/L if that's the unit you have chosen.  Is that not your experience?

**Reply by sashasochka** [reply]
Yes, it is. Unlike american (mg/dl) users who would have both axes using the same unit, it is not as easy to understand the chart, because the horizontal axis "Raw Values" uses mg/dl. What I'm also saying is that the vertical axis label explicitly specifies its unit of measurement, while the horizontal axis doesn't. 

> If the user changes their unit to mmol/L and erroneously think that the intercept limit is the same figure, it would be bad if they force the intercept.

I guess I do not entirely understand this sentence and why would the user misunderstand the figures, if they were all using the same single set of units as provided in the settings?

**Reply by PhilTU** [reply]
arguably the x-axis doesn't have to have units, a device could send a 0 to 100 output to represent its output value, and you input the corresponding glucose value to derive the calibration in your units of choice.

**Reply by Navid200** [reply]
If I look at the calibration graph, I can see that xDrip uses a linear approximation for calibration.
I can see how effective the last calibration has been considering each calibration has a weight and considering this weight is affected by the age of the calibration.

The unit of the x axis is not a part of any of those.  I mean if the unit and values of the x axis change, I can still get the same info from the graph.
The x axis really is an electrical quantity.  Hence the need for calibration to convert it to glucose.

You should be able to just ignore the values on the x axis.

Those become very important if you are a developer and need to make a change.  But, as a user, I don't understand why you care about them.

**Comment by mcoffey73** [comment]
Unsubscribe
